packages/components/src/button/button.tsx

```diff
@@ -1,12 +1,73 @@
-import { forwardRef } from 'react'
+import { cloneElement, forwardRef } from 'react'
 
 import { Stack, styled } from '@tamagui/core'
 
 import { Paragraph } from '../typography'
 
-import type { GetProps } from '@tamagui/core'
+import type { GetVariants, MapVariant, PressableProps } from '../types'
+import type { StackProps } from '@tamagui/core'
 import type { Ref } from 'react'
 
+type Variants = GetVariants<typeof Base>
+
+type Props = PressableProps & {
+  variant?: Variants['variant']
+  size?: Variants['size']
+  shape?: 'default' | 'circle'
+  children?: string
+  icon?: React.ReactElement
+  iconAfter?: React.ReactElement
+  disabled?: boolean
+}
+
+const textColors: MapVariant<typeof Base, 'variant'> = {
+  primary: '$white-100',
+  positive: '$white-100',
+  grey: '$neutral-100',
+  darkGrey: '$neutral-100',
+  outline: '$neutral-100',
+  ghost: '$neutral-100',
+  danger: '$white-100',
+}
+
+const Button = (props: Props, ref: Ref<HTMLButtonElement>) => {
+  const {
+    variant = 'primary',
+    shape = 'default',
+    size = 40,
+    icon = null,
+    iconAfter = null,
+    children,
+    ...buttonProps
+  } = props
```

Comment:
All buttons now extends `PressableProps`, which allow only the necessary handlers to be passed. Before it allowed margins, paddings, etc, which was incorrect.

Reply:
Components should limit API surface as much as possible, so the consumers won't be able to reach invalid state.

Reply:
Spacing should be done by parents using gaps.